e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js

```diff
@@ -88,4 +77,465 @@ describe('Profile popover', () => {
             cy.get('#add-user-to-channel-modal__add-button').should('be.disabled');
         });
     });
+
+    it('MM-T3 Add user — Public ON / Private OFF', () => {
+        // # Do the system console setup
+        cy.apiAdminLogin();
+        cy.visit('/admin_console/user_management/permissions/system_scheme');
```

Comment:
I see this repeated in several test blocks. Maybe move to `beforeEach`? Other that don't require it might be better to move in a separate spec file.

Reply:
Hmm .. it might look a bit weird to have T5 and T8 in a separate file and the rest in another, just to de-duplicate 2 lines. I think this is fine as is. And groups all related tests in a single file. 0/5.

Reply:
I'll defer to you but just saying for test maintainability where we spend most of the time after the script is merged. Also, on the same note, making each test isolated will help us easily debug the script in the future. You'll know it's isolated if you can run the test successfully after adding `it.only` on one `it` block.

Reply:
No worries, I'll move it out. The tests are not 100% isolated because the posts created are shared between them but the user roles being reset and now some more tests being moved out should make them stable.